pkg/controller/job/job_controller.go

```diff
@@ -1684,6 +1716,106 @@ func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
 	metrics.JobPodsFinished.WithLabelValues(completionMode, metrics.Failed).Add(float64(diff))
 }
 
+// matchPodFailurePolicy returns information about matching a given failed pod
+// against the pod failure policy rules. The information is represented as an
+// optional job termination message (present in case the pod matched a 'Terminate'
+// rule) and a boolean indicating if the failure should be counted towards
+// backoffLimit (it should not be counted if the pod matched an 'Ignore' rule).
+func matchPodFailurePolicy(job *batch.Job, failedPod *v1.Pod) (*string, bool) {
```

Comment:
nit: the job_controller.go file is pretty big. You could consider moving the functions below to a pod_failure_policy.go file.

And maybe it's worth having unit tests for just this function. That would simplify the unit tests for syncJob.

But if you think it's too late to change this before the codefreeze, leave it like this.

Reply:
Makes sense, I extracted the code with some basic tests.